### ISSUE TITLE
Classifier loss and predictor output different from primitive calculation

### ISSUE THREAD

**Issue by arisliang** [chainer/chainer#4698]
I'm following documentation (https://docs.chainer.org/en/stable/examples/cnn.html). From "Ways to calculate loss" section, it indicates that two ways of calculating loss, presumably should generate the same result.

However, it seems the first way (primitive way), and second way (Classifier way) gives different results with same seed (both way's result are consistent on different runs).

First way:
> np.random.seed(0)
model = LeNet5()
np.random.seed(0)
x = np.random.rand(32, 1, 28, 28).astype(np.float32)
np.random.seed(0)
t = np.random.randint(0, 10, size=(32,)).astype(np.int32)
y = model(x)
loss = F.softmax_cross_entropy(y, t)
print(loss)
print(y[0,0])

result:
> variable(2.3931446)
variable(0.54596496)

Second way:
> np.random.seed(0)
model = L.Classifier(LeNet5())
loss2 = model(x, t)
print(loss2)
print(model.predictor(x)[0, 0])

result:
> variable(2.4241476)
variable(0.5875003)

**Comment by arisliang** [comment]
` print(chainer.__version__)`
`print(cupy.__version__)`

> 4.0.0
4.0.0

**Comment by kmaehashi** [comment]
This is because the example is using `in_size=None` in `fc4` [Linear](https://docs.chainer.org/en/stable/reference/generated/chainer.links.Linear.html) layer:

```py
self.fc4 = L.Linear(None, 84)
```

In this case, layer parameters are initialized with the [default initializer](https://docs.chainer.org/en/stable/reference/initializers.html) e.g., `LeCunNormal`, which depends on random number. Try setting seed just before calling `model`.

**Comment by kmaehashi** [comment]
Feel free to reopen the issue if you still have any issue. Thanks!
